atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/util/ByteArrayUtilities.java

```diff
@@ -43,6 +45,21 @@ public static boolean areMapsEqual(Map<Cell, byte[]> map1, Map<Cell, byte[]> map
         return true;
     }
 
+    private static boolean areByteMapsEqual(Map<byte[], byte[]> map1, Map<byte[], byte[]> map2) {
```

Comment:
we _could_ arguably merge this with the one for cell.

Reply:
Another thing: do maps actually work with `byte[]` as keys? Can we guarantee that the key check will work too?

Reply:
as in, will `map2.containsKey(e.getKey())` ever work?

Reply:
Just checked in a test, yes it does work.

Reply:
```suggestion
    private static boolean areByteMapsEqual(SortedMap<byte[], byte[]> map1, SortedMap<byte[], byte[]> map2) {
```
Then it does work for sure. Otherwise you need to have the same reference for the key

Reply:
This is why I wouldn't merge it with the one for cell, because it doesn't work for arbitrary maps

Reply:
^ that is fair - I wanted to be abundantly careful here